### DISCUSSION TITLE
Meaning of `score` in search results in clients?

### DISCUSSION THREAD

**Discussion by ddebrunner** [milvus-io/milvus#34037] [Q&A and General discussion] [answered]
`score` as a concept is returned in a search result for at least the go-lang and Python clients.

Both document it as being the _distance_, but I think that's incorrect, isn't it returning the similarity metric used in the search?

I tested with metric.COSINE and it seems to be returning the actual metric value (which is not a distance?) with the golang client.

**Comment by yhmo** [comment, marked as answer]
search() interface always returns the similarity metric value.
For COSINE/IP(normalized), it returns the value of a1*b1 + a2*b2 + a3*b3 ... This is a similarity value, not distance.
For L2, it returns the value of (a1-b1)^2 + (a2-b2)^2 + (a3-b3)^2... This is an euclidean distance(squared).
"score" is just a general name of the similarity results.
No matter what the name it is, the topK items are ranked according to the similarity values(ascending or descending).

**Comment by ddebrunner** [comment]
@yhmo Should a doc issue be created?

**Reply by yhmo** [reply]
I think unnecessary. Both "score" or "distance" are metric of similarity. In most of time, users don't care what the name it is, they just need a topk results ranked by the score/distance in ascending or descending order.
